sealer prune failed when prune build tmp

### What happen?

When sealer image failed, try to prune the data failed by command `sealer prune`.

### Relevant log output?

```Shell
# sealer prune
start to prune image db ... 
start to prune layer ... 
3ca5ae9d0b24deabce12213b927664dc692f389dc2e826a1f301df4a4ba6a873 deleted
3ca5ae9d0b24deabce12213b927664dc692f389dc2e826a1f301df4a4ba6a873 deleted
Usage:
  sealer prune [flags]

Examples:
$ sealer prune
start to prune image db ...
start to prune layer ...
start to prune build tmp ...


Flags:
  -h, --help   help for prune

Global Flags:
      --color string               set the log color mode, the possible values can be [never always] (default "always")
      --config string              config file of sealer tool (default is $HOME/.sealer.json)
  -d, --debug                      turn on debug mode
      --hide-path                  hide the log path
      --hide-time                  hide the log time
      --log-to-file                write log message to disk
  -q, --quiet                      silence the usage when fail
      --remote-logger-url string   remote logger url, if not empty, will send log to this url
      --task-name string           task name which will embedded in the remote logger header, only valid when --remote-logger-url is set

2022-09-09 03:03:52 [ERROR] [root.go:70] sealer-v0.8.6: failed to umount target /var/lib/sealer/tmp/.DTmp-1667720254: invalid argument
```


### What you expected to happen?

pruned succeeds.

### How to reproduce it (as minimally and precisely as possible)?

```YAML
create a sealer-build failed. and make sure there is a overlay mount created by sealer. like this:

]# df -h
Filesystem               Size  Used Avail Use% Mounted on
/dev/mapper/centos-root   40G  9.6G   31G  24% /
devtmpfs                 1.9G     0  1.9G   0% /dev
tmpfs                    1.9G     0  1.9G   0% /dev/shm
tmpfs                    1.9G  8.6M  1.9G   1% /run
tmpfs                    1.9G     0  1.9G   0% /sys/fs/cgroup
/dev/mapper/centos-home   20G  259M   20G   2% /home
/dev/sda1               1014M  142M  873M  14% /boot
tmpfs                    379M     0  379M   0% /run/user/0
overlay                   40G  9.6G   31G  24% /var/lib/sealer/tmp/.DTmp-4069575571
```


### Anything else we need to know?

_No response_

### What is the version of Sealer you using?

{"gitVersion":"v0.8.6","gitCommit":"884513e","buildDate":"2022-07-12 02:58:54","goVersion":"go1.16.15","compiler":"gc","platform":"linux/amd64"}

### What is your OS environment?

CentOS 7

### What is the Kernel version?

3.10.0-862.el7.x86_64

### Other environment you want to tell us?

- Cloud provider or hardware configuration:
- Install tools:
- Others:

--- a/pkg/prune/build_pruner.go
+++ b/pkg/prune/build_pruner.go
@@ -17,7 +17,6 @@ package prune
 import (
 	"github.com/sealerio/sealer/common"
 	"github.com/sealerio/sealer/utils/mount"
-	osi "github.com/sealerio/sealer/utils/os"
 )
 
 type buildPrune struct {
@@ -33,10 +32,7 @@ func NewBuildPrune() Pruner {
 func (b buildPrune) Select() ([]string, error) {
 	var pruneList []string
 	// umount all tmp dir, and delete it
-	pruneUnits, err := osi.GetDirNameListInDir(b.pruneRootDir, osi.FilterOptions{
-		All:          true,
-		WithFullPath: true,
-	})
+	pruneUnits, err := mount.GetDirNameListInDir(b.pruneRootDir)
 	if err != nil {
 		return pruneList, err
 	}

--- a/utils/mount/mount_service.go
+++ b/utils/mount/mount_service.go
@@ -19,12 +19,12 @@ import (
 	"net"
 	"strings"
 
+	"github.com/moby/sys/mountinfo"
+	"github.com/sealerio/sealer/utils/ssh"
 	"github.com/shirou/gopsutil/disk"
 	"github.com/sirupsen/logrus"
 
-	"github.com/sealerio/sealer/utils/exec"
 	"github.com/sealerio/sealer/utils/os/fs"
-	"github.com/sealerio/sealer/utils/ssh"
 	strUtils "github.com/sealerio/sealer/utils/strings"
 )
 
@@ -120,6 +120,18 @@ func NewMountService(target, upper string, lowLayers []string) (Service, error) 
 	}, nil
 }
 
+func GetDirNameListInDir(dir string) ([]string, error) {
+	var dirs []string
+	infos, err := mountinfo.GetMounts(mountinfo.PrefixFilter(dir))
+	if err != nil {
+		return dirs, err
+	}
+	for _, info := range infos {
+		dirs = append(dirs, info.Mountpoint)
+	}
+	return dirs, nil
+}
+
 //NewMountServiceByTarget will filter file system by target,if not existed,return false.
 func NewMountServiceByTarget(target string) Service {
 	mounted, info := GetMountDetails(target)
@@ -141,12 +153,11 @@ type Info struct {
 }
 
 func GetMountDetails(target string) (bool, *Info) {
-	cmd := fmt.Sprintf("mount | grep %s", target)
-	result, err := exec.RunSimpleCmd(cmd)
+	infos, err := mountinfo.GetMounts(mountinfo.SingleEntryFilter(target))
 	if err != nil {
 		return false, nil
 	}
-	return mountCmdResultSplit(result, target)
+	return mountCmdResultSplit(infos[0].VFSOptions, target)
 }
 
 func GetRemoteMountDetails(s ssh.Interface, ip net.IP, target string) (bool, *Info) {

